iab/draft-iab-protocol-transitions.xml

```diff
@@ -913,68 +915,133 @@ Suzanne Woolf
-            </section>
+        <section title="HTTP">
+
+          <t>HTTP has been through several transitions as a protocol.</t>
+
+          <t>The first version <xref target="HTTP0.9"/> was extremely simple, with no headers,
+          status codes, or explicit versioning. HTTP/1.0 <xref target="RFC1945"/> introduced these
+          and a number of other concepts; it succeeded mostly because deployment of HTTP was still
+          relatively new, with a small pool of implementers and (comparatively) small set of
+          deployments and users.</t>
+
+          <t>HTTP/1.1 <xref target="RFC2616"/> was an attempt to make the protocol suitable for the
+          massive scale it was being deployed upon, and to introduce some new features. These
+          included:</t>
+
+          <t><list style="symbols">
+            <t>Pipelining - the ability to have more than one request to a server on the wire, to
+            mitigate Head of Line Blocking issues.</t>
+
+            <t>Expect/Continue - the ability to request that a server indicate that a request
+            header set is serviceable, so that the client can continue to send a request body with
+            reduced risk of it being rejected (thereby sacrificing the connection).</t>
+          </list></t>
+
+          <t>HTTP/2 <xref target="RFC7540"/> was largely aimed at improving performance. The
+          primary improvement was the introduction of request multiplexing, which is supported by
+          request prioritization and flow control. It also introduced header compression <xref
+          target="RFC7541"/> and binary framing; this made it completely backwards incompatible on
+          the wire, but still semantically compatible with previous versions of the protocol.</t>
+
+          <section title="Protocol Versioning, Extensions and 'Grease'">
+
+            <t>During the development of HTTP/1.1, there was a fair amount of confusing regarding
+            the semantics of HTTP version numbers, resulting in <xref target="RFC2145"/>. Later, it
+            was felt that minor versioning in a protocol caused more confusion than it was worth,
+            and so HTTP/2.0 became HTTP/2.</t>
+
+            <t>This decision was informed by the observation that many implementations ignored the
+            major version number of the protocol, or misinterpreted it. As is the case with many
+            protocol extension points, HTTP versioning had failed to be "greased" by change often
+            enough, and so had become "rusted" so that only a limited range of values could
+            interoperate.</t>
+
+            <t>This phenomenon has been observed in other protocols, such as TLS, and there are
+            active efforts to identify extension points that are in need of such "grease" so that
+            they remain flexible. This can be achieved both by good extension point design, and
+            through use of unplanned values by common implementations.</t>
```

Comment:
I disagree with the conclusion here.

A "good" design, assuming that is a reference to the IAB advice in RFC 6709, isn't a guarantee of success.  Consistent and proper implementation of the extension mechanism is all that is necessary.  There is basically only one way to guarantee that: make the extension point indispensible to implementations.

Attempts to make the extension point simple have failed, see the SNI extension in TLS.

The idea behind "greasing" is to achieve this goal by making it appear as though the extension point is being used.  Then at least implementations need to handle unknown codepoints gracefully.  Grease also attempts to make it more difficult to detect and remove greasing than it is to correctly implement the extension point.

That's a lot to capture.  I would either make the simple point - use it or lose it - or be prepared to write a whole new section in the body of the doc.